pkg/util/ratelimiter/multiratelimiter.go

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ratelimiter
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type (
+	// MultiPolicy defines the policy of a rate limiter
+	MultiPolicy struct {
+		TimeoutDuration    time.Duration
+		LimitRefreshPeriod time.Duration
+		LimitForPeriod     []int
+	}
+
+	// MultiRateLimiter defines a rate limiter
+	MultiRateLimiter struct {
+		lock      sync.Mutex
+		state     State
+		policy    *MultiPolicy
+		startTime time.Time
+		cycle     int
+		tokens    []int
+		listener  EventListenerFunc
+	}
+)
+
+// NewPolicy create and initialize a policy
+func NewMultiPolicy(timeout, refresh time.Duration, limit []int) *MultiPolicy {
```

Comment:
Please fix the Github Actions warning.

And for the prototype of the function, would `func NewMultiPolicy(timeout, refresh time.Duration, limit ...int) *MultiPolicy` be more friendly for writing the caller code?

Reply:
sounds good! and i will update `MultiRateLimiter` `AcquirePermission(counts ...int)` too. To keep these two functions in same schema.